homeassistant/components/esphome/__init__.py

```diff
@@ -304,17 +305,67 @@ async def _register_service(
 ):
     service_name = f"{entry_data.device_info.name}_{service.name}"
     schema = {}
+    fields = {}
+
     for arg in service.args:
-        schema[vol.Required(arg.name)] = {
-            UserServiceArgType.BOOL: cv.boolean,
-            UserServiceArgType.INT: vol.Coerce(int),
-            UserServiceArgType.FLOAT: vol.Coerce(float),
-            UserServiceArgType.STRING: cv.string,
-            UserServiceArgType.BOOL_ARRAY: [cv.boolean],
-            UserServiceArgType.INT_ARRAY: [vol.Coerce(int)],
-            UserServiceArgType.FLOAT_ARRAY: [vol.Coerce(float)],
-            UserServiceArgType.STRING_ARRAY: [cv.string],
+        metadata = {
+            UserServiceArgType.BOOL: {
+                "validator": cv.boolean,
+                "description": "A boolean.",
+                "example": "False",
+                "selector": {"boolean": {}},
+            },
+            UserServiceArgType.INT: {
+                "validator": vol.Coerce(int),
+                "description": "An integer number.",
+                "example": "42",
+                "selector": {"number": {}},
```

Comment:
Shouldn't this be set to box mode?

Reply:
That's weird, I see it now it [defaults to slider ](https://github.com/home-assistant/core/blob/c8ffac20b9b717b704c3c8a9d80e6fdc6b088a22/homeassistant/helpers/selector.py#L114) it did showed up in a box with up/down arrows in my test.